### ISSUE TITLE
Unexpected exception when entering invalid range in Iqt fit

### ISSUE THREAD

**Issue by RichardWaiteSTFC** [mantidproject/mantid#27440]
### Expected behavior
When entering an invalid range the programme would notify the user and ask for another range without unexpected exception or crashing mantid.

### Actual behavior
In MantidPlot 4.2 and 4.1 entering an invalid range (e.g. 5-4) produces an error in the log

![image](https://user-images.githubusercontent.com/55979119/69228151-f9052100-0b7a-11ea-817a-05d886dd404f.png)
but also produces an unexpected exception and the Error reporting window
![image](https://user-images.githubusercontent.com/55979119/69228187-091d0080-0b7b-11ea-81ac-0d6074e028ed.png)

On workbench 4.2 entering an invalid range causes a runtime error and causes the window to be greyed out and hang indefinitely (had to be closed from task manager)

![image](https://user-images.githubusercontent.com/55979119/69228239-22be4800-0b7b-11ea-8343-2c980f5e73ff.png)

### Steps to reproduce the behavior
(1) Run following code to get workspaces to fit
```
from mantid.simpleapi import *
Load(Filename='irs26174_graphite002_red.nxs', OutputWorkspace='irs26174_graphite002_red')
Load(Filename='irs26173_graphite002_res.nxs', OutputWorkspace='irs26173_graphite002_res')
TransformToIqt(SampleWorkspace='irs26174_graphite002_red', ResolutionWorkspace='irs26173_graphite002_res',
ParameterWorkspace='irs26174_graphite002_TransformToIqtParameters', OutputWorkspace='irs26174_graphite002_iqt')
```
(2) Open the Data Analysis interface > I(Q,t) tab > Multiple Input and add workspace. Enter spectrum numbers with an invalid range (e.g. 5-4)

### Platforms affected
MantidPlot and Workbench

**Comment by AnthonyLim23** [comment]
@Mantid-Matthew how important is this one?

**Comment by Harrietbrown** [comment]
Fixed by #28158
